Content.Server/Weapon/ExtraDamageAgainstWhitelistComponent.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Content.Server.Weapon.Melee;
+using Content.Shared.Whitelist;
+using Robust.Shared.Analyzers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization.Manager.Attributes;
+
+namespace Content.Server.Weapon
+{
+    /// <summary>
+    ///     Applies extra damage to
+    /// </summary>
+    [RegisterComponent, Friend(typeof(MeleeWeaponSystem))]
+    public class ExtraDamageAgainstWhitelistComponent : Component
+    {
+        public override string Name { get; } = "ExtraDamageAgainstWhitelist";
+
+        [DataField("whitelist", required: true)]
+        public EntityWhitelist Whitelist = default!;
+
+        // TODO Change to use resistanceset/damageset/whatever so this can be of arbitrary type
+
+        [DataField("damageMultiplier")]
+        public int DamageMultiplier = 1;
+
+        [DataField("flatDamage")]
+        public int FlatDamage = 0;
+    }
```

Comment:
This seems like it's just doubling up on resistance sets for damageable? I don't see the benefit over say having a window weak to piercing or whatever.

Reply:
i was going to change this once damageable ecs is in to be more in line with resistance sets, doing it like that is probably how i'll end up working it out